test/config-luatest/credentials_applier_test.lua

```diff
@@ -783,3 +783,255 @@ g.test_sync_ro_rw = function(g)
+
+            local exp_perm_2 = iconfig.credentials.users.myuser2
+            exp_perm_2 = internal.privileges_from_config(exp_perm_2)
+            exp_perm_2 = internal.privileges_add_defaults('myuser2', 'user',
+                                                          exp_perm_2)
+
+            t.assert_equals(perm_1, exp_perm_1)
+            t.assert_equals(perm_2, exp_perm_2)
+        end,
+        verify_args_2 = {iconfig},
+    })
+end
+
+g.test_postpone_grants_till_rename = function(g)
+    local iconfig = {
+        credentials = {
+            users = {
+                guest = {
+                    roles = { 'super' }
+                },
+                myuser1 = {
+                    privileges = {
+                        {
+                            permissions = {
+                                'read',
+                                'write',
+                            },
+                            spaces = {
+                                'myspace1',
+                                'myspace2',
+                            },
+                        },
+                    },
+                },
+            },
+        },
+    }
+
+    -- Create myspace1, check, rename to myspace2, check, create myspace1,
+    -- check.
+    helpers.reload_success_case(g, {
+        options = iconfig,
+        verify = function()
+            box.schema.space.create('myspace1')
+
+            local internal =
+                    require('internal.config.applier.credentials')._internal
+
+            local perm_1 = box.schema.user.info('myuser1')
+            perm_1 = internal.privileges_from_box(perm_1)
+
+            t.assert_equals(perm_1['space']['myspace1'],
+                                {read = true, write = true})
+
+            t.assert_equals(perm_1['space']['myspace2'], nil)
+
+            box.space['myspace1']:rename('myspace2')
+
+            local perm_2 = box.schema.user.info('myuser1')
+            perm_2 = internal.privileges_from_box(perm_2)
+
+            t.assert_equals(perm_2['space']['myspace2'],
+                                {read = true, write = true})
+
+            t.assert_equals(perm_2['space']['myspace1'], nil)
+        end,
+        options_2 = iconfig,
+        verify_2 = function(iconfig)
+            box.schema.space.create('myspace1')
+
+            local internal =
+                    require('internal.config.applier.credentials')._internal
+
+            local perm = box.schema.user.info('myuser1')
+            perm = internal.privileges_from_box(perm)
+
+            local exp_perm = iconfig.credentials.users.myuser1
+            exp_perm = internal.privileges_from_config(exp_perm)
+            exp_perm = internal.privileges_add_defaults('myuser1', 'user',
+                                                        exp_perm)
+            t.assert_equals(perm, exp_perm)
+        end,
+        verify_args_2 = {iconfig},
+    })
+
+    -- Create myspace2, check, rename to myspace1, check.
```

Comment:
I'm interested in effect of the following sequence of steps.

* Declare *different* myspace1 and myspace2 privs in the config.
* Create myspace1 and myspace2.
* Check that the privs for myspace1 and myspace2 in sync with the config.
* Rename myspace1 to myspace_tmp.
* Rename myspace2 to myspace1.
* Rename myspace_tmp to myspace2.
* Check that the privs for myspace1 and myspace2 in sync with the config.

I'm reading the patch starting from tests, so at the moment I don't know, whether it is important to add such a test case. Please, consider whether it is needed.

Reply:
Sigh, we discussed it verbally once again and decided to have the sync performed (per object) on every object creation or rename. So now the test can be implemented as you suggest, so I added it.